integration/sql/iface-java/src/test/java/io/openlineage/sql/OpenLineageSqlTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+/* Copyright 2018-2022 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.sql;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.Test;
+
+class OpenLineageSqlTest {
+  @Test
+  void basicParse() {
+    SqlMeta output = OpenLineageSql.parse(Arrays.asList("SELECT * FROM test")).get();
+    assertEquals(
+        output,
+        new SqlMeta(
+            Arrays.asList(new DbTableMeta(null, null, "test")),
+            new ArrayList<DbTableMeta>(),
+            Collections.emptyList()));
+  }
+
+  @Test
+  void parseWithDialect() {
+    SqlMeta output =
+        OpenLineageSql.parse(
+                Arrays.asList(
+                    "SELECT * FROM `random-project`.`dbt_test1`.`source_table` WHERE id = 1"),
+                "bigquery")
+            .get();
+    assertEquals(
+        output,
+        new SqlMeta(
+            Arrays.asList(new DbTableMeta("random-project", "dbt_test1", "source_table")),
+            new ArrayList<DbTableMeta>(),
+            Collections.emptyList()));
+  }
+
+  @Test
+  void parseError() {
+    boolean exceptionCaught = false;
+    try {
+      SqlMeta output = OpenLineageSql.parse(Arrays.asList("Definitely not an SQL statement")).get();
+    } catch (Exception e) {
+      exceptionCaught = true;
+      assertTrue(e instanceof Exception);
+    }
+    assertTrue(exceptionCaught);
+  }
+
+  @Test
+  void simpleColumnLevelLineage() {
+    SqlMeta output =
+        OpenLineageSql.parse(Arrays.asList("SELECT t1.a, b, c as x, d as y FROM table1 t1")).get();
+    assertEquals(
+        Arrays.asList(
+            columnLineage("a", "table1", "a"),
+            columnLineage("b", "table1", "b"),
+            columnLineage("x", "table1", "c"),
+            columnLineage("y", "table1", "d")),
+        output.columnLineage());
+  }
+
```

Comment:
I agree but still it may be reasonable to get from the parser information that `*` expression was used, so that we can cover this scenario when consuming parser output.

Reply:
@pawel-big-lebowski it's great comment and would be a commendable effort, but I think best left to subsequent PRs.